app/controllers/posts_controller.rb

```diff
@@ -2,4 +2,36 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
   end
+
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    @comments = @post.comments
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    if @post.save
+      flash[:success] = 'Post created successfully'
+      redirect_to roo_path
+    else
+      flash[:error] = 'The was an error while creating post'
+      render :new
+    end
+  end
+
+  def destroy
+    # to do
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```

Comment:
- [ ] You used `rails generate scaffold` to create the `User`, `Post`, `Comment` and `Like` models. The essence of building the application from the scratch is to understand the MVC pattern/framework in Ruby on Rails and how they connect to the database. Kindly start this project without using the `rails generate scaffold`. 👍 😃

Reply:
Hello @Gambit142 , as per the project requirement, i.e. ``do NOT use rails generate scaffold command.``, I didn't use scaffold command, I do not know why did you say that.